lib/pg_online_schema_change/query.rb

```diff
@@ -215,6 +216,46 @@ def storage_parameters_for(client, table)
 
         columns.first
       end
+
+      # This function acquires the lock and keeps the transaction
+      # open. If a lock is acquired, its upon the caller
+      # to call COMMIT to end the transaction. If a lock
+      # is not acquired, transaction is closed and a new transaction
+      # is started to acquire lock again
+      def lock_exclusive(client, table)
+        attempts ||= 1
+
+        client.connection.async_exec("SET lock_timeout = '#{client.wait_time_for_lock}s'; BEGIN;")
```

Comment:
it sets a `lock_timeout` which matches client.wait_time_for_lock. we use this ability for retries since on failure to acquire the lock `PG::LockNotAvailable`/`PG::InFailedSqlTransaction` is raised